src/dbus/dbus_request.hpp

```diff
@@ -43,39 +44,72 @@ class DBusRequest
         : mConnection(aConnection)
         , mMessage(aMessage)
     {
+        dbus_message_ref(aMessage);
+        dbus_connection_ref(aConnection);
     }
 
     /**
      * This method returns the message sent to call the d-bus method.
      *
-     * @returns   The dbus message
+     * @returns   The dbus message.
      */
-    SharedDBusMessage GetMessage(void) { return mMessage; }
+    UniqueDBusMessage const &GetMessage(void) { return mMessage; }
 
     /**
-     * This method replys to the d-bus method call.
+     * This method returns underlying d-bus connection.
      *
-     * @param[in] aReply  The tuple to be sent
+     * @returns   The dbus connection.
+     */
+    UniqueDBusConnection const &GetConnection(void) { return mConnection; }
+
+    /**
+     * This method replies to the d-bus method call.
+     *
+     * @param[in] aReply  The tuple to be sent.
      *
      */
     template <typename... Args> void Reply(const std::tuple<Args...> &aReply)
     {
-        DBusMessage *reply = dbus_message_new_method_return(mMessage.GetRaw());
+        UniqueDBusMessage reply(dbus_message_new_method_return(mMessage.get()));
```

Comment:
I was wondering why a function call of `reply` has to declare the return type :(

Should we prefer C++11 `list initialization` to make it distinguishable?

```C++
 UniqueDBusMessage reply {dbus_message_new_method_return(mMessage.get())};
```

Reply:
The return type is void. The `template <typename... Args>` should be in a new line but the auto format seems to have its own idea.

Reply:
Updated to prefer list initialization.